iModelJsNodeAddon/JsInterop.cpp

```diff
@@ -990,14 +992,19 @@ BentleyStatus JsInterop::ConvertSchemas(bvector<Utf8String> const& inputStrings,
             bool conversionStatus = ECSchemaConverter::Convert(*const_cast<ECSchemaP> (schema), *schemaContext);
             if (!conversionStatus)
                 return BentleyStatus::ERROR;
-            Utf8String schemaXml;
-            SchemaWriteStatus writeStatus = schema->WriteToXmlString(schemaXml);
-            if (SchemaWriteStatus::Success != writeStatus)
+            for (int i = 0; i < schemaKeyPairs.size(); i++)
                 {
-                outputStrings.clear();
-                return BentleyStatus::ERROR;
+                if (schemaKeyPairs[i].first.Matches(schema->GetSchemaKey(), SchemaMatchType::Exact))
+                    {
+                    SchemaWriteStatus writeStatus = schema->WriteToXmlString(outputStrings[i]);
+                    if (SchemaWriteStatus::Success != writeStatus)
+                        {
+                        outputStrings.clear();
+                        return BentleyStatus::ERROR;
+                        }
+                    break;
+                    }
                 }
-            outputStrings.push_back(schemaXml);

```

Comment:
Don't you still need to populate the outputStrings vector somewhere?

Reply:
Indeed, I'm resizing the `outputStrings` before iterating through the schemas vector. Following this, `schema->WriteToXmlString(outputStrings[i])` method is used to populate the `outputStrings` vector.